david_laster/class2/my_calc.rb

```diff
@@ -0,0 +1,44 @@
+puts "What would you like to do? ('add', 'subtract', 'multiply', 'divide')?"
+operation = gets.chomp
+puts "enter the first number"
+num1 = gets.chomp.to_i
+puts "enter the second number"
+num2 = gets.chomp.to_i
+
+def add(x,y)
+	x + y
+end
+
+def subtract(x,y)
+	x - y
+end
+
+def multiply(x,y)
+	x * y
+end
+
+def divide(x,y)
+	x / y
+end
+
+def modular(x,y)
+	x % y
+end
+
+def calc(operation, num1, num2)
```

Comment:
well done, sir!

Since that's such a long conditional statement - you could even tackle this with a case statement! Check it out:

```
def calc(operation, num1, num2)
  case
  when operation == "add"
    add(num1,num2)
  when operation == "subtract"
    subtract(num1,num2)
  when operation == "multiply"
    multiply(num1,num2)
  when operation == "divide"
    divide(num1,num2)
  else
    "unknown without a valid operation"
  end
end
```

It may or may not be what you want in this situation but it's always good to be aware of different approaches. You can read more about the case statement here: http://www.skorks.com/2009/08/how-a-ruby-case-statement-works-and-what-you-can-do-with-it/